RFC/src/RFC-0241_AtomicSwapXMR.md

```diff
@@ -0,0 +1,789 @@
+and publish the Monero transaction containing the Monero UTXO with the aggregate key \\(Xm = Xm_a + Xm_b \\).
+
+### Claim XTR 
+
+When Alice sees that the Monero UTXO that Bob created is mined on the Monero blockchain containing the correct aggregate 
+key \\(Xm\\), she can provide Bob with the requird `pre_image` to spend the Tari UTXO. She does not have the 
+missing key \\(x_b \\) to claim the Monero yet, but it will be revealed when Bob claims the Tari. 
+
+Bob can now supply the `pre_image` and his Monero private key as transaction input to unlock the script.
+
+### Claim XMR
+
+Alice can now see that Bob spent the Tari UTXO, and by examining the `input_data` required to satisfy the script, she 
+can learn Bob's secret Monero key. Although this private key is now public knowledge, her part of the Monero spend key 
+is still private, and thus only she knows the complete Monero spend key. She can use this knowledge to claim the Monero 
+UTXO.
+
+### The refund
+
+If something goes wrong and Bob never publishes the Monero, or he disappears, Alice needs to wait for the lock height
+`height_1` to pass. This will allow her to reclaim her Tari, but in doing so, she needs to publish her Monero secret key 
+as input to the script to unlock the Tari. When Bob comes back online, he can use this public knowledge to reclaim his 
+Monero as only he now knows both parts of the Monero UTXO spend key.
+
+
+### The lapse transaction
+
+If something goes wrong and Alice never gives Bob the requird `pre_image`, Bob needs to wait for the lock height
+`height_2` to pass. This will allow him to claim the Tari he wanted all along, but in doing so, he needs to publish
+his Monero secret key as input to the script to unlock the Tari. When Alice comes back online, she can use this public 
+knowledge to claim the Monero she wanted all along as only she now knows both parts of the Monero UTXO spend key.
+
+
+## Method 2
+
+### Detail
+
+This method uses adapter signatures and multi-party commitment signatures to ensure that the spending
+party leaks their private Monero key part. Because all keys are aggregates keys, we need to ensure that the refund
+and lapse transactions are negotiated and signed before Alice publishes the Tari UTXO. This will allow either Alice or Bob to
+claim the refund and lapse transactions, respectively, without the other party being online. 
+
+### TariScript
+
+The Script used for the Tari UTXO is as follows:
+``` TariScript,ignore
+   CheckHeight(height_1)
+   LtZero
+   IFTHEN
+      PushPubkey(K_{Ss})
+   Else
+      CheckHeight(height_2)
+      LtZero
+      IFTHEN
+         PushPubkey(K_{Sr})
+      Else
+         PushPubkey(K_{Sl})
+      ENDIF
+   ENDIF
+```
+
+Before `height_1`, Bob can claim the Tari UTXO if Alice gives him the correct signature to complete the transaction.
+After `height_1` but before `height_2`, Alice can claim the Tari UTXO. After `height_2,` Bob can claim the Tari UTXO.
+
```

Comment:
I think we need a short discussion about choosing safe values for height_1 and height_2, similar to the previous comment.

Reply:
See comments above, they should be the same and only discussed once.

Reply:
Yes, it could just be mentioned here again, i.e. "_... as previously discussed ..._".